packages/controls/src/controls/definition.ts

```diff
@@ -74,6 +79,20 @@ export abstract class ControlDefinition<
     return data as Data
   }
 
+  resolveValue(
+    _data: DataType | undefined,
+  ): Resolvable<ResolvedValueType | undefined> {
+    console.assert(
+      false,
+      `${this.controlType}: 'resolveValue' is not implemented`,
+    )
```

Comment:
Default implementation in the abstract `ControlDefinition` class should never be called. All basic controls provide an override of this method.